galsim/compound.py

```diff
@@ -175,6 +174,69 @@ def __str__(self):
         return '(' + ' + '.join(str_list) + ')'
         #return 'galsim.Sum([%s])'%', '.join(str_list)
 
+    def _prepareDraw(self):
+        for obj in self._obj_list:
+            obj._prepareDraw()
+        SBList = [obj.SBProfile for obj in self._obj_list]
+        self.SBProfile = galsim._galsim.SBAdd(SBList, self._gsparams)
+
+    def shoot(self, n_photons, rng=None):
+        """Shoot photons into a PhotonArray.
+
+        @param n_photons    The number of photons to use for photon shooting.
+        @param rng          If provided, a random number generator to use for photon shooting,
+                            which may be any kind of BaseDeviate object.  If `rng` is None, one
+                            will be automatically created, using the time as a seed.
+                            [default: None]
+        @returns PhotonArray.
+        """
+        # Setup the rng if not provided one.
+        if rng is None:
+            ud = galsim.UniformDeviate()
+        elif isinstance(rng, galsim.BaseDeviate):
+            ud = galsim.UniformDeviate(rng)
+        else:
+            raise TypeError("The rng provided is not a BaseDeviate")
+
+        remainingAbsoluteFlux = self.SBProfile.getPositiveFlux() + self.SBProfile.getNegativeFlux()
+        fluxPerPhoton = remainingAbsoluteFlux / n_photons
+
+        # Initialize the output array
+        result = galsim._galsim.PhotonArray(n_photons)
+
+        remainingN = n_photons
+        istart = 0  # The location in the result array where we assign the component arrays.
+
+        # Get photons from each summand, using BinomialDeviate to randomize
+        # the distribution of photons among summands
+        for i, obj in enumerate(self.obj_list):
+            thisAbsoluteFlux = obj.SBProfile.getPositiveFlux() + obj.SBProfile.getNegativeFlux()
+
+            # How many photons to shoot from this summand?
+            thisN = remainingN  # All of what's left, if this is the last summand...
+            if i < len(self.obj_list)-1:
+                # otherwise, allocate a randomized fraction of the remaining photons to summand.
+                bd = galsim.BinomialDeviate(ud, remainingN, thisAbsoluteFlux/remainingAbsoluteFlux)
+                thisN = bd()
+            if thisN > 0:
```

Comment:
I noticed the False version of this branch isn't covered.  Probably worth adding tests of shooting 0,1,2 to the `do_shoot` function to make sure they run without errors, even though the results won't be particularly checkable for accuracy.